include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -77,12 +83,102 @@ namespace picongpu
              * medium. Since it is mostly used to analyze electron bunches, this plugin
              * assumes that the analyzed particles have the mass and charge of electrons.
              *
-             * @tparam T_ParticlesType particle type to compute transition radiation from
+             * @tparam ParticlesType particle type to compute transition radiation from
              */
-            template<typename T_ParticlesType>
-            class TransitionRadiation : public ILightweightPlugin
+            template<typename ParticlesType>
+            class TransitionRadiation : public plugins::multi::IInstance
             {
             private:
+                struct Help : public plugins::multi::IHelp
+                {
+                    /** creates an instance
+                     *
+                     * @param help plugin defined help
+                     * @param id index of the plugin, range: [0;help->getNumPlugins())
+                     */
+                    std::shared_ptr<IInstance> create(
+                        std::shared_ptr<IHelp>& help,
+                        size_t const id,
+                        MappingDesc* cellDescription) override
+                    {
+                        return std::shared_ptr<IInstance>(
+                            new TransitionRadiation<ParticlesType>(help, id, cellDescription));
+                    }
+
+                    //! periodicity of computing the particle energy
+                    plugins::multi::Option<std::string> notifyPeriod
+                        = {"period", "enable plugin [for each n-th step]"};
+                    plugins::multi::Option<std::string> optionFileName
+                        = {"file", "file name to store transition radiation in: ", "transRad"};
+                    plugins::multi::Option<std::string> optionFileExtention
+                        = {"ext",
+                           "openPMD filename extension. This controls the"
+                           "backend picked by the openPMD API. Available extensions: ["
+                               + openPMD::printAvailableExtensions() + "]",
+                           openPMD::getDefaultExtension().c_str()};
+                    plugins::multi::Option<bool> optionTextOutput
+                        = {"datOutput",
+                           "optional output: transition radiation as text file in readable format for the "
+                           "in picongpu provided python analysis script, 1==enabled",
+                           0};
+                    plugins::multi::Option<float_X> foilPositionSI
```

Comment:
The naming should reflect that the position is for the Y-direction.

Reply:
done